### PR TITLE
WIP: chore(fieldset): remove serializedGroup specs

### PR DESCRIPTION
**Pull request by jorenbroekema**
Notes for implementation:
- __serializeElements / __serializeElement / serializeGroup methods should be removed, that's the purpose of this. Just the serializedValue getter in LionFieldset should be enough.
- The modelValue of choice groups changes in https://github.com/ing-bank/lion/pull/507 and will affect the serializedValue as well.

### CODE REVIEW

File: packages/fieldset/test/lion-fieldset.test.js [hunk 1]
```diff
@@ -614,24 +614,20 @@ describe('<lion-fieldset>', () => {
...
       <${tag}>
         <${childTag} name="price"></${childTag}>
+        <${childTag} name="firstName"></${childTag}>
       </${tag}>`);
       await nextFrame();
       fieldset.formElements.price.modelValue = 0;
-      expect(fieldset.serializeGroup()).to.deep.equal({ price: 0 });
-    });
-
-    it('__serializeElements serializes 0 as a valid value', async () => {
```
Comment (jorenbroekema) [inline packages/fieldset/test/lion-fieldset.test.js]: Unnecessary test, and this method will probably be removed as it's only used by serializeGroup and not necessary for serializedValue

File: packages/fieldset/test/lion-fieldset.test.js [hunk 2]
```diff
@@ -614,24 +614,32 @@ describe('<lion-fieldset>', () => {
...
+      const fieldset = await fixture(html`
+      <${tag}>
+        <${childTag} name="price"></${childTag}>
+        <${childTag} name="firstName"></${childTag}>
+      </${tag}>`);
       await nextFrame();
-      const elements = [{ serializedValue: 0 }];
-      expect(fieldset.__serializeElements(elements)).to.deep.equal([0]);
+      fieldset.formElements.price.modelValue = null;
+      fieldset.formElements.firstName.modelValue = undefined;
```
Comment (thatianamg) [inline packages/fieldset/test/lion-fieldset.test.js]: Re-evaluate undefined, see what the platform does, might make sense to include this.

Comment (thatianamg) [reply packages/fieldset/test/lion-fieldset.test.js]: See also separate Unparseable story

**Comment by daKmoR** [comment]
maybe that was a profound misunderstanding but do we want to the `serializedValue` to be able to sync from/to a database?
```
<form>
  <fielset disabled name="fullName">
    <input name="firstName">
    <input name="lastName">
  </fieldset>
  <input name="age">
```

now the modelValue will be
```
{
  fullName: {
    firstName: '...',
    lastName: '...',
  },
  age: 10
}
```

but if we combine serializeGroup and serializedValue e.g. by not doing including disabled dates it would just be
```
{
  age: 10
}
```

e.g. enabling if at some point the fieldset gets enabled... it will mean the firstName/lastName will not have the database values...

but then again - we do not have a disabled state anywhere in the modelValue, serializedValue so maybe it's fine... and you need to manage anything disabled separately?

**Comment by daKmoR** [comment]
will be addressed by https://github.com/ing-bank/lion/pull/583

**Pull request closed without merging**